runtime/caml/compatibility.h

```diff
@@ -56,10 +56,6 @@
 /* **** array.c */
 
 /* **** backtrace.c */
-#define backtrace_active CAML_DEPRECATED("backtrace_active", "caml_backtrace_active") caml_backtrace_active
```

Comment:
Make mental note: remove "compatibility mode" entirely before 5.00 is released.

Reply:
Given that we will remove compatibility mode for 5.00, it would be a good time to make `CAML_NAME_SPACE` defined by default. See https://github.com/ocaml/ocaml/issues/4013. This would also simplify the usages of `CAML_NAME_SPACE` in `domain_state.h`.